src/main/java/org/xstefank/webhook/WebHookEndpoint.java

```diff
@@ -44,18 +47,27 @@ public void processPullRequest(JsonNode payload) {
     }
 
     private static FormatConfig readConfig() {
-        String configFileName = System.getProperty(TEMPLATE_FORMAT_FILE);
-        if (configFileName == null) {
-            configFileName = System.getProperty(Utils.JBOSS_CONFIG_DIR) + "/format.yaml";
-        }
-        log.info(configFileName);
-        File configFile = new File(configFileName);
+        String url = getUrl();
         ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-
         try {
+            if (url != null)
+                return mapper.readValue(new URL(url).openStream(), FormatConfig.class);
+            String configFileName = getTemplateFileName();
+            log.info(configFileName);
+            File configFile = new File(configFileName);
             return mapper.readValue(configFile, FormatConfig.class);
         } catch (IOException e) {
             throw new IllegalArgumentException("Cannot load configuration file", e);
         }
     }
+
+    private static String getUrl() {
+        String url = System.getProperty(TEMPLATE_FORMAT_URL);
+        return url == null ? Utils.getTemplateUrlFromProperties() : url;
+    }
+
+    private static String getTemplateFileName() {
```

Comment:
if you chose to include url in the properties configuration too, you should also align potential path in case it's not defined by the system property